### DISCUSSION TITLE
Calculating load using FLORIS

### DISCUSSION THREAD

**Discussion by conorfhayes** [NatLabRockies/floris#414] [Q&A] [answered]
Hi,

I am trying to calculate both power generated by the turbines and also the load on the turbines. I know CCBlade can be used to calculate load but is there any function to calculate load in FLORIS?

Any help with this would be great!

**Comment by Bartdoekemeijer** [comment, marked as answer]
Hi @conorfhayes,

FLORIS is not able to calculate the turbine loads due to its steady-state and wind-farm-oriented nature. Its intended to be used as a lightweight wind farm model that can be used for applications like layout optimization. Accurate load calculations require dynamic models of the turbine structure and inflow wind field at a high time resolution, like CCBlade. Though, CCBlade is only for a single wind turbine.

Literature exists where a large precalculated set of the turbine damage-equivalent loads, such as from CCBlade or NRELs [OpenFAST](https://github.com/OpenFAST/openfast), is connected to a steady-state wind farm model like FLORIS. You can find such applications in the literature:

1. [Van Dijk et al.](https://www.sciencedirect.com/science/article/pii/S0360544217300518)
2. [Mendes Reyez et al.](https://wes.copernicus.org/articles/4/549/2019/)

Alternative "lightweight" methods for calculating loads also exist, like [Stanley et al.](https://wes.copernicus.org/articles/7/433/2022/wes-7-433-2022.pdf).

This is just a set of articles that I recall from memory, but I am confident that there are more. Again, these are all somewhat simplified methods for calculating the loads. Accurate wind farm power and load simulations require computational fluid dynamics simulations using, e.g., NRELs [SOWFA](https://github.com/NREL/SOWFA). Though, those simulations also require hundreds of CPU hours for an hour of simulation of a small farm and are typically difficult for those unfamiliar with CFD and large-eddy simulation to set up.
